src/nvhttp.cpp

```diff
@@ -267,12 +268,32 @@ namespace nvhttp {
 
     launch_session.host_audio = host_audio;
     launch_session.gcm_key = util::from_hex<crypto::aes_t>(get_arg(args, "rikey"), true);
+    std::stringstream mode = std::stringstream(get_arg(args, "mode"));
+    // Split mode by the char "x", to populate width/height/fps
+    int x = 0;
+    std::string segment;
+    while (std::getline(mode, segment, 'x')) {
+      if (x == 0) launch_session.width = stoi(segment);
```

Comment:
`std::stoi()` throws when the conversion can't be performed, but not sure if it can happen here. Maybe use `atoi(segment.c_str())` from C since it can't throw.